### DISCUSSION TITLE
Getting 502 Bad Gateway

### DISCUSSION THREAD

**Discussion by gagan-ncompass** [actix/actix-web#2397] [Q&A]
`2021/10/08 07:31:25 [error] 602#602: *7273 upstream prematurely closed connection while reading response header from upstream,`

I have been getting this error for some of endpoints. I am using nginx and actix-web 3.3.2. 

I referred to [https://stackoverflow.com/questions/36488688/nginx-upstream-prematurely-closed-connection-while-reading-response-header-from](url) and increased the timeout, but I am still getting the same error.

I am thinking that the issue has something to do with my actix-web server.

Any help would be deeply appreciated
Thank you

**Comment by aliemjay** [comment]
How to reproduce?

**Reply by gagan-ncompass** [reply]
@aliemjay I have about 30 endpoints running on default actix-web HttpServer and nginx configuration

**Comment by kogai** [comment]
The same error was observed in our environment while using actix-web 4.0.1.

We don't yet know how to reproduce it, and we are setting the following environment variables in our production environment:
RUST_LOG=info,actix-web=debug.

So, let's observe the logs for a while 👁️.
